src/platform/nm-linux-platform.c

```diff
@@ -3506,6 +3514,43 @@ _new_from_nl_qdisc (struct nlmsghdr *nlh, gboolean id_only)
 	obj->qdisc.parent = tcm->tcm_parent;
 	obj->qdisc.info = tcm->tcm_info;
 
+	if (tb[TCA_OPTIONS]) {
+		struct nlattr *options_attr;
+		int remaining;
+
+		nla_for_each_nested (options_attr, tb[TCA_OPTIONS], remaining) {
+			if (nla_len (options_attr) < sizeof (uint32_t))
+				continue;
+
+			switch (nla_type (options_attr)) {
+			case TCA_FQ_CODEL_LIMIT:
```

Comment:
I think we should check that the kind is "fq_codel" before parsing options as TCA_FQ_CODEL_* because other qdiscs use the same numerical attribute values.

Reply:
True. Will fix.